p2p/peer.go

```diff
@@ -364,7 +480,8 @@ func (p *peer) metricsReporter() {
 	for {
 		select {
 		case <-metricsTicker.C:
-			status := p.mconn.Status()
+			// TODO: this is a bit of a hack, we should have a better way to get the status.
```

Comment:
This is indeed terrible. The best way I can see is to create a `p2p/status` package and declare the common fields there.

Reply:
done ✅